### ISSUE TITLE
Successful coveralls runs but failure reporting back to GitHub

### ISSUE THREAD

**Issue by andy-maier** [lemurheavy/coveralls-public#1565]
We are experiencing this issue every now and then. The latest occurrence can be seen in this PR: https://github.com/pywbem/pywbemtools/pull/1002

The symptom is always that each of the parallel GitHub Actions workflow runs within the PR reports coverage successfully to coveralls.io, e.g. from https://github.com/pywbem/pywbemtools/pull/1002/checks?check_run_id=2987125999:

```
  if ! [[ 3.9 =~ (3.4) ]]; then coveralls; fi
  shell: /usr/bin/bash -l {0}
  env:
    TEST_SERVER_IMAGE: keyporttech/smi-server:0.1.2
    TEST_SERVER_IMAGE_TAR: smi-server-0.1.2.tar.gz
    DOCKER_CACHE_DIR: ~/docker-cache
    pythonLocation: /opt/hostedtoolcache/Python/3.9.5/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.9.5/x64/lib
    GITHUB_TOKEN: ***
    COVERALLS_PARALLEL: true
    COVERALLS_FLAG_NAME: ubuntu-latest,3.9,latest
Submitting coverage to coveralls.io...
Coverage submitted!
Job #1000022063.4
https://coveralls.io/jobs/83054666
```

But coveralls.io fails to report back to GitHub, as can be seen in the PR:
<img width="854" alt="Screenshot 2021-07-05 at 08 14 05" src="https://user-images.githubusercontent.com/8939758/124425600-12b3a600-dd69-11eb-8208-af8b1c30e6df.png">

This causes the overall status of the PR on GitHub to be failed. Even though we can merge such PRs, it is still annoying.

The problem comes and goes. From the past few occurrences, it does not only show up on PRs with 0% coverage change, but also for small coverage changes. See the Closed PR history of the project. Nearly every failed closed PR failed due to this issue.

Note that the coveralls.io status page says all systems operational at this point.

**Comment by andy-maier** [comment]
The coverage threshold settings seem to be the root cause for this behavior.

The current coverage of the master branch of this repo is at 82%, but the coveralls.io threshold settings were at 90% absolute with 1% decrease. Changing the threshold settings to nothing for the absolute value and 5% decrease resulted in successful reports back to GitHub.

I discovered this just by chance, since there was no mentioning of the reason for the failure anywhere. It would be great if the reason for the failure could be made visible somehow. Ideally of course in the comment that is added to the PR.

The comment from coveralls.io on a PR that failed due to the threshold settings, says just this:
```
Coverage remained the same at 81.747% when pulling 0ddfa80 on andy/docs-listener into 6836936 on master.
```

**Comment by afinetooth** [comment]
@andy-maier thanks for the update. The first thing I checked were the coverage threshold settings and when I saw no value for COVERAGE THRESHOLD FOR FAILURE, figured we had a genuine issue. I'm glad this was the reason and that it was resolvable for you.

Point taken on the failure message. I've submitted an enhancement request to include failure reason, particularly in the case where a failure tracks to the settings. Will try to surface this better in updated docs too.

Closing for now. Just reply / re-open if necessary.
